src/compiler/Stage/Parse/Parser.elm

```diff
@@ -419,30 +419,30 @@ literalNumber =
             |= parseLiteralNumber
         , parseLiteralNumber
         ]
-        |> P.inContext InLiteralNumber
+        |> P.inContext InNumber
 
 
-{-| for literalChar and, in the future, literalString
--}
-character : Parser_ Char
-character =
+character : Maybe Char -> Parser_ Char
```

Comment:
shouldn't it be a List of Char ? So that we can also exclude new lines for double-quotes strings ?

Reply:
Possibly yeah, related to https://github.com/elm-in-elm/compiler/pull/46/files#diff-27fa2f98c2b4d7c0696e05c001f6c1c6R519

Reply:
Something like (quick and dirty):

```type Quotes
    = SingleQuote
    | DoubleQuote
    | TrippleQuote

character : Quotes -> Parser_ String
character quotes =
    P.oneOf
        [ P.succeed identity
            |. P.token (P.Token "\\" ExpectingEscapeBackslash)
            |= P.oneOf
                [ P.map (\_ -> '\\') (P.token (P.Token "\\" (ExpectingEscapeCharacter '\\')))
                , P.map (\_ -> '"') (P.token (P.Token "\"" (ExpectingEscapeCharacter '"'))) -- " (elm-vscode workaround)
                , P.map (\_ -> '\'') (P.token (P.Token "'" (ExpectingEscapeCharacter '\'')))
                , P.map (\_ -> '\n') (P.token (P.Token "n" (ExpectingEscapeCharacter 'n')))
                , P.map (\_ -> '\t') (P.token (P.Token "t" (ExpectingEscapeCharacter 't')))
                , P.map (\_ -> '\u{000D}') (P.token (P.Token "r" (ExpectingEscapeCharacter 'r')))
                , P.succeed identity
                    |. P.token (P.Token "u" (ExpectingEscapeCharacter 'u'))
                    |. P.token (P.Token "{" ExpectingUnicodeEscapeLeftBrace)
                    |= unicodeCharacter
                    |. P.token (P.Token "}" ExpectingUnicodeEscapeRightBrace)
                ]
            |> P.inContext InCharEscapeMode
        -- Exclude chars, depending on Quotes
        , P.andThen (\_ -> P.problem ExpectingClosing) <|
            case quotes of
                SingleQuote ->
                    P.oneOf
                        [ P.token (P.Token "\n" (ExpectingEscapeCharacter 'n'))
                        , P.token (P.Token "'" ExpectingClosing)
                        ]

                DoubleQuote ->
                    P.oneOf
                        [ P.token (P.Token "\n" (ExpectingEscapeCharacter 'n'))
                        , P.token (P.Token "\"" ExpectingClosing)
                        ]

                TrippleQuote ->
                    P.token (P.Token "\"\"\"" ExpectingClosing)

        , P.succeed identity
            |= P.getChompedString (P.chompIf (Just >> (/=) stoppingDelimiter) ExpectingChar)
            |> P.andThen
                (\string ->
                    string
                        |> String.uncons
                        |> Maybe.map (Tuple.first >> P.succeed)
                        |> Maybe.withDefault (P.problem (CompilerBug "Multiple characters chomped in `character`"))
                )
        ]```

Reply:
Thanks, when I next work on this I'll definitely try that approach!